Modules/Loadable/Models/SubjectHierarchyPlugins/qSlicerSubjectHierarchyModelsPlugin.cxx

```diff
@@ -191,32 +192,37 @@ QString qSlicerSubjectHierarchyModelsPlugin::tooltip(vtkIdType itemID)const
     return QString("Error");
   }
 
-  // Get basic tooltip from abstract plugin
-  QString tooltipString = Superclass::tooltip(itemID);
-
   vtkMRMLModelNode* modelNode = vtkMRMLModelNode::SafeDownCast(shNode->GetItemDataNode(itemID));
-  vtkPolyData* polyData = modelNode ? modelNode->GetPolyData() : nullptr;
-  vtkMRMLModelDisplayNode* displayNode = modelNode ? vtkMRMLModelDisplayNode::SafeDownCast(modelNode->GetDisplayNode()) : nullptr;
-  if (modelNode && displayNode && polyData)
+  if (modelNode == nullptr)
   {
-    bool visible = (displayNode->GetVisibility() > 0);
-    tooltipString.append( QString(" (Points: %1  Cells: %2  Visible: %3")
-      .arg(polyData->GetNumberOfPoints()).arg(polyData->GetNumberOfCells())
-      .arg(visible ? "YES" : "NO") );
-    if (visible)
-    {
-        double color[3] = {0.0,0.0,0.0};
-        displayNode->GetColor(color);
-      tooltipString.append( QString("  Color: %4,%5,%6  Opacity: %7%")
-        .arg(int(color[0]*255.0)).arg(int(color[1]*255.0)).arg(int(color[2]*255.0))
-        .arg(int(displayNode->GetOpacity()*100.0)) );
-    }
-    tooltipString.append(QString(")"));
+    return QString("Invalid model node");
   }
-  else
+  vtkPointSet* data = modelNode->GetPolyData();
+  if (data == nullptr)
+  {
+    data = modelNode->GetUnstructuredGrid();
+  }
+  if (data == nullptr)
+  {
+    return QString("Model node does not contain data");
+  }
+
+  // Get basic tooltip from abstract plugin
+  QString tooltipString = Superclass::tooltip(itemID);
+  tooltipString.append( QString(" (Points: %1  Cells: %2  Visible: %3")
+    .arg(data->GetNumberOfPoints()).arg(data->GetNumberOfCells())
+    .arg(modelNode->GetDisplayVisibility() > 0 ? "YES" : "NO") );
```

Comment:
Do we really need the display visibility? Isn't the eye icon enough? If not enough then we should maybe add it at higher levels, for all nodes. I would also consider just adding "Not visible." if not visible ("Visible: yes/no" adds quite a lot of noise).

Reply:
I was thinking about this as well, but I decided to keep the scope on the unstructured grid level (and the basic readability because that error message was really strange).

I can remove the visibility part, no problem. It should be obvious what is not visible. What I wanted to avoid is changing more plugins, but I'll look in case others have this too.